### PR TITLE
[Merged by Bors] - chore(Algebra): use newly introduced finset notation

### PR DESCRIPTION
**Pull request by YaelDillies**
For `s : Finset α`, replace `s.card` with `#s`, `s.filter p` with `{x ∈ s | p x}`, `∑ x ∈ s.filter p, f x` with `∑ x ∈ s with p x, f x`. Rewrap lines around and open the `Finset` locale where necessary.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Group/Finset.lean [hunk 1]
```diff
@@ -929,11 +929,11 @@ theorem prod_eq_mul {s : Finset α} {f : α → β} (a b : α) (hn : a ≠ b)
         prod_const_one
 
 -- Porting note: simpNF linter complains that LHS doesn't simplify, but it does
-/-- A product over `s.subtype p` equals one over `s.filter p`. -/
+/-- A product over `s.subtype p` equals one over `{x ∈ s | p x}`. -/
 @[to_additive (attr := simp, nolint simpNF)
-  "A sum over `s.subtype p` equals one over `s.filter p`."]
+  "A sum over `s.subtype p` equals one over `{x ∈ s | p x}p`."]
```
Comment (loefflerd) [inline Mathlib/Algebra/BigOperators/Group/Finset.lean:934]: typo

File: Mathlib/Algebra/Order/BigOperators/Group/Finset.lean [hunk 2]
```diff
@@ -175,38 +175,38 @@ lemma mul_le_prod {i j : ι} (hf : ∀ i ∈ s, 1 ≤ f i) (hi : i ∈ s) (hj :
...
   calc
-    (∏ y ∈ t, ∏ x ∈ s.filter fun x ↦ g x = y, f x) ≤
-        ∏ y ∈ t ∪ s.image g, ∏ x ∈ s.filter fun x ↦ g x = y, f x :=
+    (∏ y ∈ t, ∏ x ∈ s with g x = y, f x) ≤
+        ∏ y ∈ t ∪ s.image g, ∏ x ∈ s with g x = y, f x :=
       prod_le_prod_of_subset_of_one_le' subset_union_left fun y _ ↦ h y
     _ = ∏ x ∈ s, f x :=
       prod_fiberwise_of_maps_to (fun _ hx ↦ mem_union.2 <| Or.inr <| mem_image_of_mem _ hx) _
 
 -- Porting note: Mathport warning: expanding binder collection (y «expr ∉ » t)
```
Comment (loefflerd) [inline Mathlib/Algebra/Order/BigOperators/Group/Finset.lean:205]: this is presumably no longer relevant now

**Review by loefflerd — COMMENTED** [review]
LGTM. I found one tiny typo, and also noticed a porting note that can be removed. 

(It is really great to see how much more natural the arguments look with the new notation!)

File: Mathlib/Algebra/Order/BigOperators/Group/Finset.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/BigOperators/Group/Finset.lean]: Looks like it stopped being relevant a while ago, but yes well spotted

**Comment by loefflerd** [comment]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Pull request closed without merging**